src/synchronous.js

```diff
@@ -5,18 +5,22 @@ var Promise = require('./core.js');
 module.exports = Promise;
 Promise.enableSynchronous = function () {
   Promise.prototype.isPending = function() {
-    return this._state == 0;
+    return this.getState() == 0;
   };
 
   Promise.prototype.isFulfilled = function() {
-    return this._state == 1;
+    return this.getState() == 1;
   };
 
   Promise.prototype.isRejected = function() {
-    return this._state == 2;
+    return this.getState() == 2;
   };
 
   Promise.prototype.getValue = function () {
+    if (this._state === 3 && this._value instanceof Promise) {
```

Comment:
You shouldn't need the instanceof check here. The state is only set to 3 if the promise is resolved with another promise.